tests/test_package.py

```diff
@@ -234,3 +236,40 @@ def test_submit_duplicate_package(
 
     else:
         assert all(scan.status != Status.QUEUED for scan in test_data)
+
+
+@pytest.mark.parametrize(
+    "name, version, score, queued_by, reported_by, queued_at",
+    [
+        ("pyfoo", "0.2.1", 13, "Ryan", "Ryan", datetime(2024, 3, 5, 12, 30, 0)),
+        ("pybar", "2.83.2", 0, "Bob", None, datetime(2024, 3, 5, 12, 30, 0)),
+        ("pybaz", "0.1.0", 56, "Daniel", None, datetime(2024, 3, 5, 12, 30, 0)),
+    ],
+)
+def test_package_from_db(
+    name: str,
+    version: str,
+    score: Optional[int],
+    queued_by: str,
+    reported_by: Optional[str],
+    queued_at: datetime,
+):
+    """Test the from_db method of Package."""
+
+    scan = Scan(
+        name=name,
+        version=version,
+        score=score,
+        queued_by=queued_by,
+        reported_by=reported_by,
+        queued_at=queued_at,
+    )
+
+    pkg = (Package.from_db(scan)).model_dump()
+
+    assert pkg.get("queued_at") == int(queued_at.timestamp())
+    assert pkg.get("name") == name
+    assert pkg.get("version") == version
+    assert pkg.get("score") == score
+    assert pkg.get("queued_by") == queued_by
+    assert pkg.get("reported_by") == reported_by
```

Comment:
I don't believe this makes a functional difference but you could just use `Package.from_db(scan)` without calling `.model_dump()` on it and accessing the object normally. Might cut down on duplicated code.

Reply:
It does make a difference, as the fields are only serialized after you dump it as per Pydantic's documentation.
If you don't, `datetime` isn't capable of being asserted as a POSIX timestamp

Reply:
ah, gotcha. looks like you're trying to test two things at once, then (that the datetime fields are serialized into a timestamp properly, and also that the general structure of `Package` is what you expect). it might be worth breaking these two tests apart: one test that tests the structure (what you have now, but don't worry about `.model_dump()`, just compare to a datetime object), and another test that is specifically concerned with the 3 (or 4?) datetime models. There, you will want to `.model_dump`, then compare to `int(queued_at.timestamp()` (similar to what you are doing now, but for all the datetime fields)

Reply:
I see, got it.
Would you be comfortable with transforming this mark;
```py
@pytest.mark.parametrize(
    "name, queued_at, pending_at, finished_at, reported_at",
    [
        ("pyfoo", datetime(2023, 2, 1, 12, 42, 1), datetime(2023, 9, 1, 16, 33, 0), datetime(2024, 1, 3, 9, 56, 46),
         datetime(2022, 12, 31, 23, 56, 0)),
        ("pybar", datetime(2023, 3, 5, 8, 10, 15), datetime(2023, 10, 12, 13, 45, 30), datetime(2024, 2, 15, 18, 20, 5),
         datetime(2022, 11, 20, 5, 30, 0)),
        ("pybaz", datetime(2023, 4, 10, 16, 20, 30), datetime(2023, 11, 20, 21, 15, 45),
         datetime(2024, 3, 25, 6, 40, 20), datetime(2023, 1, 15, 12, 0, 0)),
    ],
)
``` 
Into a `pytest.fixture`? Then we could do; 
```py
@pytest.fixture
def scan_data():
    return [
        {
            "name": "pyfoo",
            "version": "0.12.1",
            "queued_at": datetime(2023, 2, 1, 12, 42, 1),
            "pending_at": datetime(2023, 9, 1, 16, 33, 0),
            "finished_at": datetime(2024, 1, 3, 9, 56, 46),
            "reported_at": datetime(2022, 12, 31, 23, 56, 0),
        },
        {
            "name": "pybar",
            "version": "3.12.2",
            "queued_at": datetime(2023, 3, 5, 8, 10, 15),
            "pending_at": datetime(2023, 10, 12, 13, 45, 30),
            "finished_at": datetime(2024, 2, 15, 18, 20, 5),
            "reported_at": datetime(2022, 11, 20, 5, 30, 0),
        },
        {
            "name": "pybaz",
            "version": "0.24.8",
            "queued_at": datetime(2023, 4, 10, 16, 20, 30),
            "pending_at": datetime(2023, 11, 20, 21, 15, 45),
            "finished_at": datetime(2024, 3, 25, 6, 40, 20),
            "reported_at": datetime(2023, 1, 15, 12, 0, 0),
        },
    ]

# ...

@pytest.mark.parametrize("scan_data", range(3))
def test_datetime_serialization(scan_data):
    ...
```

I find the test data much more readable as a fixture. Thoughts?

Reply:
Moreover, scan fixtures could provide the advantage of standardization across multiple tests. Would you be open to considering this approach?

Reply:
I would rather not parametrize here at all, to be honest. Test parametrization is most valuable when used in situations such as these: https://github.com/vipyrsec/dragonfly-mainframe/blob/08460a2a0de6b13be43f19cd48ef8b8ba64bd70e/tests/test_package.py#L28-L36
Where we can test edge cases.
In this case, there aren't really "edge cases" and we're just testing the same thing but with different values for the datetimes and whatnot. I'd rather just test one thing like so:
```py
def test_package_from_db():
    scan = Scan(
        name="abc",
        version="def",
        queued_at=datetime(2024, 3, 5, 12, 30, 0),
    )

    pkg = Package.from_db(scan)

    assert pkg.name == "abc"
    assert pkg.version == "version"
    assert pkg.queued_at == datetime(2024, 3, 5, 12, 30, 0)
    # maybe test one or two more fields just to be sure but it's not too important
```

And then testing the timestamp conversions in another test:
```py
def test_package_serializes_datetime():
    pkg = Package(..., queued_at=datetime(2024, 3, 5, 12, 30, 0))
    serialized = pkg.model_dump()
    
    assert serialized.get("queued_at") == 1709663400
    assert serialized.get("reported_at") == ...
    ... # you get the idea
````

Overall I don't think parametrization nets us much value here as we'd simply be testing different values of the same thing, and the code itself is really just passing values around and not manipulating it too much.

Reply:
I see, that's valid thinking. I'll adapt and commit shortly.